.github/workflows/dist.yml

```diff
@@ -73,18 +73,24 @@ jobs:
         uses: hendrikmuhs/ccache-action@v1
         with:
           create-symlink: true
-          key: ccache-${{ runner.os }}-${{ runner.arch }}
+          key: ccache-sdist-${{ runner.os }}-${{ runner.arch }}
           evict-old-files: "7d"
 
       - name: Test SDist buildable
+        # We extract sdist to same folder (./build) and install package from there
+        # to make sure ccache don't get a lot of miss because of different working dir.
```

Comment:
This might be caused by the GHA cache-key collision. You should add relevant information to the cache key to avoid collection.

Reply:
Since the ccache problem is due to the malformed GHA cache. You will not need to manually untar the SDist and do a manual build below. Just pip install from the SDist.

Reply:
`pip install sdist.tar.gz` will extract source in different tmp dir and will make ccache (sccache as well) not able to cache compilation.

Reply:
`CCACHE_NOHASHDIR` looks helpful, trying

Reply:
I guess we can set the following in `CMakeLists.txt`:

```cmake
set(CMAKE_C_COMPILER_LAUNCHER "${CCACHE_PROGRAM};base_dir=${CMAKE_SOURCE_DIR}" CACHE STRING "C compiler launcher")
set(CMAKE_CXX_COMPILER_LAUNCHER "${CCACHE_PROGRAM};base_dir=${CMAKE_SOURCE_DIR}" CACHE STRING "CXX compiler launcher")
set(CMAKE_CUDA_COMPILER_LAUNCHER "${CCACHE_PROGRAM};base_dir=${CMAKE_SOURCE_DIR}" CACHE STRING "CUDA compiler launcher")
```